arch/powerpc/kernel/eeh_sysfs.c

```diff
@@ -91,6 +91,23 @@ static ssize_t eeh_pe_state_store(struct device *dev,
 
 static DEVICE_ATTR_RW(eeh_pe_state);
 
+#ifdef CONFIG_PCI_IOV
```

Comment:
root@z2533ap19:~# cat /sys/class/pci_bus/0015\:01/device/0015\:01\:00.0/eeh_notify_resume
0x00000000
^^ pf 

root@z2533ap19:~# cat /sys/class/pci_bus/0015\:01/device/0015\:01\:00.2/eeh_notify_resume
0x00000000

^^ vf
root@z2533ap19:~#

Reply:
Where is your eeh_notify_resume_store operation? It is very unconventional to have your configfs read op go and actually perform something that would do more than retrieve and return a state. For example configfs checks different file permissions on a read call than a write call. So if your op is supposed to be a read only which in this case is _show then you should basically just be returning a state without making calls to perform work that should technically be done under a write permission context. This is all for security purposes and I think having your _show do more than just getting a state opens up doors for exploits. If you look at examples of any usage of device attributes no one really does work within the _show op. Its all just returning a state stored somewhere.

Reply:
"This is all for security purposes and I think having your _show do more than just getting a state opens up doors for exploits. If you look at examples of any usage of device attributes no one really does work within the _show op"

^^^ -->> don't understand this if that was the case then why does eeh_pe_state_show does that?

so what you are saying you want me to store the return value? then just output it back to the user in query? You understand the implications of that ... I will have to go and store all values and then report them out? I did not want to diverge in that design ... but i can .. not what i wanted to do, though.

Reply:
@bgly  thanks for the comments, here is the output. Let me know what you think.

224.558127] pseries_send_allow_unfreeze: Failed to allow unfreeze for PHB#15-PE#10000, rc=fffffffd
root@z2533ap19:~# cat  /sys/class/pci_bus/0015\:01/device/0015\:01\:00.1/eeh_notify_resume
-3
root@z2533ap19:~# echo 32 > /sys/class/pci_bus/0015\:01/device/0015\:01\:00.1/sriov_numvfs

root@z2533ap19:~# echo 1 > /sys/class/pci_bus/0015\:01/device/0015\:01\:00.1/eeh_notify_resume

root@z2533ap19:~# cat  /sys/class/pci_bus/0015\:01/device/0015\:01\:00.1/eeh_notify_resume
0